### PR TITLE
Add docker-compose-yml

### PR DESCRIPTION
**Pull request by leinadsened**
Add Docker Compose for running front- and back-end After running "docker-compose up --build" the app should be available on localhost:8080

### CODE REVIEW

**Comment by leinadsened** [comment]
CLOSES #96

**Comment by ashmilhussain** [comment]
I tried running this, but I’m encountering an error.
Screenshot 2024-10-08 at 10 44 23 AM


In the case of Docker Compose, we already know the backendUrl. How should we handle this?

We don’t need an environment variable in this case, but it shouldn’t result in a failure either.

@agberoz  have a look

**Comment by leinadsened** [comment]
I've found the problem, I have this version of the ui dockerfile:
image
Can the frontend's dockerfile be modified or should I update the docker-compose?

**Comment by sethumadhavan-k** [comment]
@leinadsened Set the environment variable as a build argument in the Dockerfile and pass it through the Docker Compose file.

**Comment by leinadsened** [comment]
Hey @sethumadhavan-k I've updated both files, please let me know what you think.
I might misunderstand something, but shouldn't "VITE_API_URL=http://backend:8001" be added to the sample .env file?

**Review by agberoz — CHANGES_REQUESTED** [review]

**Review by sethumadhavan-k — DISMISSED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - PYTHONUNBUFFERED=1
+    command: >
+      bash -c "source /opt/venv/bin/activate && python main.py --config ./config.yaml llm"
+
+  frontend:
+    build:
+      context: ./ui
+      dockerfile: Dockerfile
+      args:
+        - VITE_API_URL=${VITE_API_URL}
```
[the remark this answers is not in the document]
Comment (leinadsened) [reply docker-compose.yml]: Updated both files

**Review by agberoz — APPROVED** [review]

**Pull request merged**